File: web/app/api/models/spaces/project.rb
module App
  class Api
    module Models
      module Spaces
        class Project < Models::SpacesItem

          def to_h
            {identifier: identifier}.merge(YAML.load(blueprint.to_yaml).to_h)
          end

          def blueprint
            Spaces.universe.blueprints.by(descriptor)
          end

          def create
            # TODO: create project
            identifier
          end

          def delete
            Spaces.universe.blueprints.delete(blueprint)
          end

          def installations
            Installations.new(self)
          end

        end
      end
    end
  end
end

File: web/app/client/app/spaces/blueprints/project.js
app.spaces.blueprints.blueprint = controller => (a,x) => [
  a.h2( controller.params.blueprint_id ),
  controller.routes( {
    '/?': app.spaces.blueprints.show,
    '/~delete': app.spaces.blueprints.delete,
    '/resolutions*': app.spaces.blueprints.resolutions,
  } ),
]
